core/src/main/java/com/mentorting/aem/core/services/RetouchedWorkflowProcess.java

```diff
@@ -0,0 +1,70 @@
+package com.mentorting.aem.core.services;
+
+import com.adobe.granite.asset.api.Asset;
+import com.adobe.granite.workflow.WorkflowSession;
+import com.adobe.granite.workflow.exec.WorkItem;
+import com.adobe.granite.workflow.exec.WorkflowProcess;
+import com.adobe.granite.workflow.metadata.MetaDataMap;
+import org.apache.sling.api.resource.ModifiableValueMap;
+import org.apache.sling.api.resource.PersistenceException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+import org.osgi.framework.Constants;
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Component(
+        service = WorkflowProcess.class,
+        property = {
+                Constants.SERVICE_DESCRIPTION + "=Retouched Metadata Workflow Process",
+                Constants.SERVICE_VENDOR + "=Mentoring",
+                "process.label=Retouched Workflow Process Metadata value setting"
+        })
+public class RetouchedWorkflowProcess implements WorkflowProcess {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    public static final String RETOUCHED = "retouched";
+    public static final String JPEG_FORMAT = "image/jpeg";
+
+    @Override
+    public void execute(WorkItem workItem, WorkflowSession workflowSession, MetaDataMap metaDataMap){
+        try(ResourceResolver resourceResolver = workflowSession.adaptTo(ResourceResolver.class)) {
+            if(resourceResolver == null){
+                logger.error("Resource Resolver is null");
+                return;
+            }
+            String payload = workItem.getWorkflowData().getPayload().toString();
+            String baseAssetPath = payload.split("/renditions")[0];
+            Resource assetResource = resourceResolver.getResource(baseAssetPath);
+            if(assetResource == null){
+                logger.error("Resource is null");
+                return;
+            }
+            Asset asset = assetResource.adaptTo(Asset.class);
+            if(asset == null){
+                logger.error("Asset is null");
+                return;
+            }
+            Resource metadataResource = assetResource.getChild("jcr:content/metadata");
+            if(metadataResource == null){
+                logger.error("Metadata Resource is null");
+                return;
+            }
+            ModifiableValueMap assetMetadata = metadataResource.adaptTo(ModifiableValueMap.class);
+            if(assetMetadata == null){
```

Comment:
In production code, you wouldn't need so many null-checks because these objects that the workflow operates with are not null in 99%. Chaining of Optional map() would be fine